app/builders/campaigns/campaign_conversation_builder.rb

```diff
@@ -0,0 +1,35 @@
+class Campaigns::CampaignConversationBuilder
+  pattr_initialize [:contact_inbox_id!, :campaign_display_id]
+
+  def perform
+    @contact_inbox = ContactInbox.find(@contact_inbox_id)
+    @campaign = @contact_inbox.inbox.campaigns.find_by!(display_id: campaign_display_id)
+
+    # We won't send campaigns if a conversation is already present
+    return if @contact_inbox.conversations.present?
+
+    ActiveRecord::Base.transaction do
+      @conversation = ::Conversation.create!(conversation_params)
+      Messages::MessageBuilder.new(@campaign.sender, @conversation, message_params).perform
+    end
+    @conversation
+  end
+
+  private
+
+  def message_params
+    ActionController::Parameters.new({
+                                       content: @campaign.message
+                                     })
+  end
+
+  def conversation_params
+    {
+      account_id: @campaign.account_id,
+      inbox_id: @contact_inbox.inbox_id,
+      contact_id: @contact_inbox.contact_id,
+      contact_inbox_id: @contact_inbox.id,
+      campaign_id: @campaign.id
+    }
```

Comment:
Should we add browser params here also? That might be relevant right?

Reply:
Product question: Should we create the conversation in resolve status by default? Because this would be shown in the UI and might be a noise right?

Reply:
@pranavrajs  browser params are now passed into the builder,
let's take the starting conversation in resolved status as a setting in our next iteration.